openedx/core/djangoapps/profile_images/views.py

```diff
@@ -81,11 +95,19 @@ def post(self, request, username):
                         status=status.HTTP_400_BAD_REQUEST,
                     )
 
+                # come up with a unique version for the uploaded pic
+                new_version = _make_image_version()
+
                 # generate profile pic and thumbnails and store them
-                create_profile_images(uploaded_file, get_profile_image_names(username))
+                create_profile_images(uploaded_file, get_profile_image_names(username, new_version))
+
+                # if there was previously a different image, delete it
+                old_version = get_profile_image_version(username)
+                if old_version is not None:
+                    remove_profile_images(get_profile_image_names(username, old_version))
 
                 # update the user account to reflect that a profile image is available.
-                set_has_profile_image(username, True)
+                set_profile_image_version(username, new_version)
```

Comment:
As mentioned at stand up, this also opens up a window where a browser can be handed the old filename but by the time it makes the request the file has gone.

How about we include the timestamp only in the URL as a query parameter? This isn't what I was proposing on Friday, but it seems that it might be better. The browser will be able to cache it indefinitely just the same, but we don't have the two issues enumerated here. 

http://stackoverflow.com/questions/23603023/file-caching-query-string-vs-last-modified

I think this also addresses my earlier comment about garbage collection of old profile images, where the process needs to know what file names to look for.